aten/src/ATen/core/ATenDispatch.cpp

```diff
@@ -7,4 +7,32 @@ ATenDispatch & globalATenDispatch() {
   return singleton;
 }
 
+void* ATenOpTable::getFallbackOp(TensorTypeId tid) const {
+  // TODO: an alternate strategy here would be to mask out the dead key
+  // and then redispatch gain (automatic delegation).  I haven't done this
+  // for now to make it easier to smoke out error cases.
+  if (function_table_[static_cast<int64_t>(TensorTypeId::UndefinedTensorId)] == nullptr) {
+    // If there is no fallback dispatch, and dispatch failed because we didn't
+    // find any valid keys to dispatch on, this usually means the user gave
+    // us a non-empty list of tensors.  So report a better error in this case.
+    // TODO: Maybe we should reword this error message
+    if (tid == TensorTypeId::UndefinedTensorId) {
+      TORCH_CHECK(false, "expected a non-empty list of Tensors")
```

Comment:
Once we're in this branch, perf doesn't matter anymore. You say that an empty list of tensors is usually the reason for this, but doesn't always have to be. The error message can be very misleading if it actually wasn't an empty list of tensors. Can we add an if/else here and show a correct error message?

Reply:
As the code is currently structured, this unfortunately can't be conveniently done. I will have to think about how to restructure this area so I can get a good error message in this case :/

Reply:
Perf only doesn't matter if we assume the user doesn't use try/except as part of a serious pipeline.